Dynatrace/dynatrace-operator
# Fixes DynaKube phase not changed correctly

# Description

If the ActiveGate reconciler determines that there are no changes to be made, it returns a boolean value of `false`.
This caused the dynakube reconciliation loop to return prematurely and to not reconcile other parts such as the phase.

These changes only return in the case of an error.

## How can this be tested?

* Deploy the operator with a cloud native dynakube that also has activegate capabilities enabled
* The phase of the applied dynakube should change correctly to `Running` after all components have become ready

## Checklist
